tests/src/main.cpp

```diff
@@ -38,7 +38,9 @@ int main(int argc, char **argv)
         chatterino::NetworkManager::deinit();
 
         settingsDir.remove();
-        QApplication::exit(res);
+        // This should be QApplication::exit(res);
+        // but using this will deadlock in ~QHostInfoLookupManager (if the twitch account was changed)
+        _exit(res);
```

Comment:
This is actually really sad, but I don't see a way to avoid this without refactoring network-request internals.

Reply:
Ok, I kinda lied. There is one way around it: calling `NetworkManager::workerThread.terminate()` after quitting the thread, which is probably a bit less ugly but still ugly.

Reply:
Hmm, the `terminate` doesn't seem to be that reliable. Delaying the loading of the 7TV user-id by 500ms seems to work. It's not 100% ideal, but fine since the ID is only needed once a message is sent.